evalml/automl/engine/engine_base.py

```diff
@@ -149,6 +157,14 @@ def train_and_score_pipeline(
     start = time.time()
     cv_data = []
     logger.info("\tStarting cross validation")
+    threshold_tuning_objective = automl_config.objective
+    if (
+        is_binary(automl_config.problem_type)
+        and automl_config.optimize_thresholds
+        and automl_config.objective.score_needs_proba
+        and automl_config.alternate_thresholding_objective is not None
+    ):
+        threshold_tuning_objective = automl_config.alternate_thresholding_objective
```

Comment:
@bchen1116 why is this code necessary? You've already delegated this complexity down into `train_pipeline`! That's called here in `train_and_score_pipeline` below, so all you need to do is provide the automl config to `train_pipeline` unmodified.

Reply:
Update: ah I see you need the threshold tuning objective for a log message. I'm leaving another suggestion about that below

Reply:
Ah, that was to try and log the threshold properly, but I think I can get rid of it and do a different check!